src/monaca.js

```diff
@@ -759,6 +759,84 @@
    * @method
    * @memberof Monaca
    * @description
+   *   Checks if the project can be built.
+   * @return {Promise}
+   * @example
+   *   monaca.checkBuildAvailability('myProjectID', 'android', 'debug').then(
+   *     function() {
+   *       //Build the project
+   *     },
+   *     function(err) {
+   *       //Cannot build the project
+   *     }
+   *   );
+   */
+  Monaca.prototype.checkBuildAvailability = function(projectId, platform, buildType) {
+    var deferred = Q.defer();
+
+    if (typeof platform == 'undefined') {
+      deferred.resolve();
+      return deferred.promise;
+    } else if (typeof buildType === 'undefined') {
+      deferred.reject(new Error("Missing --build-type parameter."));
+      return deferred.promise;
+    }
+
+    this._get('/project/' + projectId + '/can_build_app')
+    .then(
+      function(response) {
+        try {
+          var data = JSON.parse(response);
+        } catch (err) {
+          return deferred.reject(new Error(JSON.stringify(err)));
```

Comment:
Please just reject the Error object, and handle it properly on the receiver side.